src/json.py

```diff
@@ -0,0 +1,223 @@
+"""json.py
```

Comment:
Question: Was this custom written, or sourced from somewhere else?

Reply:
Custom written using the syntax diagrams from https://www.json.org/.

Reply:
a8bd3104fdda2c74b4f76ed88574bc1bfffe9eb0